Index: my_vector.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- my_vector.cpp	(date 1525896286000)
+++ my_vector.cpp	(date 1525896286000)
@@ -41,7 +41,11 @@
 
 
 size_t my_vector::size() const {
-    return is_small() ? small_object.size() : big_object.size();
+    if (is_small()) {
+        return small_object.size();
+    } else {
+        return size_;
+    }
 }
 
 
@@ -74,14 +78,23 @@
 
 void my_vector::pop_back() {
     try_split();
-    is_small() ? small_object.pop_back() : big_object.pop_back();
+    if (is_small()) {
+        small_object.pop_back();
+    } else {
+        --size_;
+    }
 }
 
 
 void my_vector::push_back(uint const &value) {
     try_split();
     ensure_cap(size() + 1);
-    is_small() ? small_object.push_back(value) : big_object.push_back(value);
+    if (is_small()) {
+        small_object.push_back(value);
+    } else {
+        big_object.ptr->long_ob[size_] = value;
+        ++size_;
+    }
 }
 
 
@@ -137,7 +150,7 @@
         clear();
         big_object.ptr = static_cast<info *>(operator new(sizeof(info) + sizeof(uint) * n * 2));
         big_object.ptr->use_count = 1;
-        big_object.size_ = sz;
+        size_ = sz;
         big_object.b_capacity = n * 2;
         //
         for (size_t i = 0; i < sz; ++i) {
